runtime/src/chpl-file-utils.c

```diff
@@ -158,6 +158,39 @@ qioerr chpl_fs_is_link(int* ret, const char* name) {
   return 0;
 }
 
+qioerr chpl_fs_is_mount(int* ret, const char* name) {
+  qioerr err = 0;
+  struct stat nBuf, parentBuf;
+  int exitStatus = 0;
+  size_t nameLen = strlen(name);
+  char* parent = (char* ) chpl_mem_allocManyZero(nameLen + 4, sizeof(char), CHPL_RT_MD_OS_LAYER_TMP_DATA, 0, 0);
```

Comment:
You can save the teensiest amount of time here by using chpl_mem_allocMany() (no "Zero"), since the next thing you're going to do is to store into every byte of the string.